Modules/Configurables/Controllers/PerspectiveController.cs

```diff
@@ -1,11 +1,103 @@
-﻿using Microsoft.AspNetCore.Http;
+﻿using AppraisalTracker.Modules.AppraisalActivity.Models;
+using AppraisalTracker.Modules.AppraisalActivity.Services;
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 
 namespace AppraisalTracker.Modules.AppraisalActivity.Controllers
 {
-    [Route("api/[controller]")]
+    [Route("perspective")]
     [ApiController]
     public class PerspectiveController : ControllerBase
     {
+        private readonly IConfigMenuItemService _configMenuItemService;
+
+        public PerspectiveController(IConfigMenuItemService configMenuItemService)
+        {
+            _configMenuItemService = configMenuItemService;
+        }
+
+        // Retrieves all perspectives
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<ConfigMenuItem>>> GetPerspectives()
+        {
+            var perspectives = await _configMenuItemService.FetchPerspectives();
+            return Ok(perspectives);
+        }
+
+        // Retrieves a single perspective
+        [HttpGet("{id}")]
+        public async Task<ActionResult<ConfigMenuItem>> GetPerspective(Guid id)
+        {
+            try
+            {
+                var perspective = await _configMenuItemService.FetchPerspective(id);
+                return Ok(perspective);
+            }
+            catch (KeyNotFoundException)
+            {
+                return NotFound();
+            }
+        }
+
+        // Adds a perspective
+        [HttpPost]
+        public async Task<ActionResult<ConfigMenuItem>> AddPerspective([FromBody] ConfigMenuItem perspective)
+        {
+            try
+            {
+                var newPerspective = await _configMenuItemService.AddConfigMenuItem(perspective);
+                return CreatedAtAction(nameof(GetPerspective), new { id = newPerspective.ItemId }, newPerspective);
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+
+        // Updates a perspective
+        [HttpPost("{id}")]
+        public async Task<ActionResult<ConfigMenuItem>> UpdatePerspective(Guid id, [FromBody] ConfigMenuItem perspective)
+        {
+            if (id != perspective.ItemId)
+            {
+                return BadRequest("ID mismatch.");
+            }
+
+            try
+            {
+                var updatedPerspective = await _configMenuItemService.UpdatePerspective( id,perspective);
+                return Ok(updatedPerspective);
```

Comment:
Move Error and exception handling to the methods in the services.
Controller is congested.